system_metrics_collector/test/topic_statistics_collector/test_subscriber_topic_statistics.cpp

```diff
@@ -0,0 +1,394 @@
+   *
+   * @return data streucture containing the collectors
+   */
+  std::vector<std::shared_ptr<topic_statistics_collector::TopicStatisticsCollector<
+      sensor_msgs::msg::Imu>>> GetCollectors() const
+  {
+    return statistics_collectors_;
+  }
+
+private:
+  /**
+   * Test publish for the fixture.
+   */
+  void PublishStatisticMessage() override
+  {
+    ++times_published_;
+  }
+
+  std::atomic<int> times_published_{0};
+};
+
+/**
+ * Node to publish messages on the test topic to trigger subscriber callbacks.
+*/
+class ImuMessagePublisher : public rclcpp::Node
+{
+public:
+  ImuMessagePublisher()
+  : Node("imu_publisher"), publisher_(nullptr)
+  {
+    publisher_ = create_publisher<sensor_msgs::msg::Imu>(kTestTopicName, 10);
+  }
+
+  ~ImuMessagePublisher() = default;
+
+  /**
+   * Publish a  single message.
+   */
+  void publish(std::shared_ptr<sensor_msgs::msg::Imu> msg)
+  {
+    publisher_->publish(*msg);
+  }
+
+  uint64_t getSubscriptionCount() const
+  {
+    return publisher_->get_subscription_count();
+  }
+
+private:
+  typename rclcpp::Publisher<sensor_msgs::msg::Imu>::SharedPtr publisher_;
+};
+
+/**
+ * Test fixture to test a SubscriberTopicStatisticsNode node
+ */
+class SubscriberTopicStatisticsNodeTestFixture : public testing::Test
+{
+public:
+  void SetUp() override
+  {
+    rclcpp::init(0, nullptr);
+
+    rclcpp::NodeOptions options;
+    options.append_parameter_override(kPublishPeriodParam, kDontPublishDuringTest.count());
+    options.append_parameter_override(kCollectStatsTopicName, kTestTopicName);
+
+    imu_publisher_ = std::make_shared<ImuMessagePublisher>();
+    EXPECT_EQ(imu_publisher_->getSubscriptionCount(), 0);
+
+    test_periodic_publisher_ = std::make_shared<TestSubscriberTopicStatisticsNode>(
+      kTestNodeName, options);
+
+    const auto collectors = test_periodic_publisher_->GetCollectors();
+    ASSERT_FALSE(collectors.empty());
+
+    moving_average_statistics::StatisticData data;
+    for (const auto & collector : collectors) {
+      data = collector->GetStatisticsResults();
+      ASSERT_TRUE(std::isnan(data.average));
+      ASSERT_TRUE(std::isnan(data.min));
+      ASSERT_TRUE(std::isnan(data.max));
+      ASSERT_TRUE(std::isnan(data.standard_deviation));
+      ASSERT_EQ(0, data.sample_count);
+    }
+  }
+
+  void TearDown() override
+  {
+    test_periodic_publisher_->shutdown();
+    EXPECT_EQ(State::PRIMARY_STATE_FINALIZED, test_periodic_publisher_->get_current_state().id());
+    EXPECT_FALSE(test_periodic_publisher_->IsPublisherActivated());
+
+    test_periodic_publisher_.reset();
+    imu_publisher_.reset();
+    rclcpp::shutdown();
+  }
+
+protected:
+  // this test is not designed to have the statistics published and reset at any point of the test,
+  // so this is defining the publish period to be something amply larger than the test duration
+  // itself
+  static constexpr std::chrono::milliseconds kDontPublishDuringTest = 2 * kTestDuration;
```

Comment:
I am not understanding this variable name. It appears to be a time duration, but it sounds like a boolean. Is there a more descriptive name to use?